browser/ui/webui/brave_new_tab_message_handler.cc

```diff
@@ -264,10 +280,22 @@ void BraveNewTabMessageHandler::OnJavascriptAllowed() {
   pref_change_registrar_.Add(kNewTabPageShowGemini,
     base::Bind(&BraveNewTabMessageHandler::OnPreferencesChanged,
     base::Unretained(this)));
+
+  if (brave::IsTorProfile(profile_)) {
+    tor_profile_service_ =
+        TorProfileServiceFactory::GetForProfile(profile_);
+    DCHECK(tor_profile_service_);
+    tor_profile_service_->AddObserver(this);
+    auto data =
+        GetTorPropertiesDictionary(tor_profile_service_->IsTorConnected(), "");
+    FireWebUIListener("tor-tab-data-updated", data);
```

Comment:
This _probably_ works ok timing-wise, but probably best to have a `getTorProperties` function, similar to all the other data types.

Reply:
fixed and force-pushed https://github.com/brave/brave-core/pull/4802/commits/26b53fae74acfe3e670c9737c0ee708eb5d4541b